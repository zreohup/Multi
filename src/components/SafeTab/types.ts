export interface safeTabItem<T> {
  label: string
  Component: React.FC<T>
}
