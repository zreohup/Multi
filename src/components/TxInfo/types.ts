import { Transaction } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
import { TxType } from '@/src/hooks/useTransactionType'

export type TxCardPress = { tx: Transaction; type?: TxType }
