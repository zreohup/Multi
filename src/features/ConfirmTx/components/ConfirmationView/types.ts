import { SettingsChangeTransaction } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
import { Address } from '@/src/types/address'

// TODO: fix it in the @safe-global/store/gateway/AUTO_GENERATED types
export type NormalizedSettingsChangeTransaction = SettingsChangeTransaction & {
  settingsInfo: SettingsChangeTransaction['settingsInfo'] & {
    owner: { value: Address; name: string }
    threshold: number
  }
}
