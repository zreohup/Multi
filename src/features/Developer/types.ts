export type Info = {
  device: {
    brand: string
    deviceName: string
    manufacturer: string
    modelId: string
    modelName: string
    osBuildId: string
    osInternalBuildId: string
    osName: string
    osVersion: string
  }
  application: {
    applicationName: string
    applicationId: string
    applicationVersion: string
    applicationBuildNumber: string
    gatewayUrl: string
    fcmToken: string
  }
}
