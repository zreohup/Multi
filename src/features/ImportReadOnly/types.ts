import { z } from 'zod'
import { formSchema } from '@/src/features/ImportReadOnly/schema'
import { useLazySafesGetOverviewForManyQuery } from '@safe-global/store/gateway/safes'

type LazyQueryResult = ReturnType<typeof useLazySafesGetOverviewForManyQuery>[1]

export type FormValues = z.infer<typeof formSchema> & {
  importedSafeResult?: {
    data: LazyQueryResult['data']
    isFetching: LazyQueryResult['isFetching']
    error: LazyQueryResult['error']
  }
}
