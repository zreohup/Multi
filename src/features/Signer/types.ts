import { z } from 'zod'
import { formSchema } from '@/src/features/Signer/schema'

export type FormValues = z.infer<typeof formSchema>
