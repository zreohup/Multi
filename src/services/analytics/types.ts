import type {
  Transaction,
  TransferTransactionInfo,
  SettingsChangeTransaction,
} from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
export type {
  Transaction,
  TransferTransactionInfo,
  SettingsChangeTransaction,
  CustomTransactionInfo,
  SwapTransferTransactionInfo,
} from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
import type { ANALYTICS_LABELS } from './constants'

/**
 * Firebase Analytics Event Types
 * These event names are passed directly to Firebase Analytics
 */
export enum EventType {
  SCREEN_VIEW = 'screen_view',
  CLICK = 'customClick',
  META = 'metadata',
  SAFE_APP = 'safeApp',
  SAFE_CREATED = 'safe_created',
  SAFE_ACTIVATED = 'safe_activated',
  SAFE_OPENED = 'safe_opened',
  WALLET_CONNECTED = 'wallet_connected',
  TX_CREATED = 'tx_created',
  TX_CONFIRMED = 'tx_confirmed',
  TX_EXECUTED = 'tx_executed',
}

export type EventLabel = string | number | boolean | null

export type AnalyticsEvent = {
  eventName: string
  eventCategory: string
  eventAction: string
  eventLabel?: EventLabel
  chainId?: string
}

export enum AnalyticsUserProperties {
  WALLET_LABEL = 'walletLabel',
  WALLET_ADDRESS = 'walletAddress',
}

// Extract precise types from the source of truth
export type TransactionInfoType = Transaction['txInfo']['type']
export type TransferInfoType = TransferTransactionInfo['transferInfo']['type']
export type SettingsInfoType = SettingsChangeTransaction['settingsInfo']['type']

// Union of all possible analytics labels
export type AnalyticsLabel =
  | (typeof ANALYTICS_LABELS.BASE_TYPES)[TransactionInfoType]
  | (typeof ANALYTICS_LABELS.TRANSFER_TYPES)[TransferInfoType]
  | (typeof ANALYTICS_LABELS.SETTINGS_TYPES)[SettingsInfoType]
  | (typeof ANALYTICS_LABELS.ENHANCED)[keyof typeof ANALYTICS_LABELS.ENHANCED]
