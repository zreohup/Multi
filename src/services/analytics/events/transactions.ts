import { EventType, type AnalyticsLabel } from '../types'

const TRANSACTIONS_CATEGORY = 'transactions'

export const TRANSACTIONS_EVENTS = {
  CONFIRM: {
    eventName: EventType.TX_CONFIRMED,
    eventCategory: TRANSACTIONS_CATEGORY,
    eventAction: 'Confirm transaction',
    // eventLabel can be set when tracking the event using AnalyticsLabel
  },
  CREATE: {
    eventName: EventType.TX_CREATED,
    eventCategory: TRANSACTIONS_CATEGORY,
    eventAction: 'Create transaction',
    // eventLabel can be set when tracking the event using AnalyticsLabel
  },
  EXECUTE: {
    eventName: EventType.TX_EXECUTED,
    eventCategory: TRANSACTIONS_CATEGORY,
    eventAction: 'Execute transaction',
    // eventLabel can be set when tracking the event using AnalyticsLabel
  },
}

// Helper function to create a transaction confirm event with a specific label
export const createTxConfirmEvent = (label: AnalyticsLabel) => ({
  ...TRANSACTIONS_EVENTS.CONFIRM,
  eventLabel: label,
})

// Helper function to create a transaction create event with a specific label
export const createTxCreateEvent = (label: AnalyticsLabel) => ({
  ...TRANSACTIONS_EVENTS.CREATE,
  eventLabel: label,
})

// Helper function to create a transaction execute event with a specific label
export const createTxExecuteEvent = (label: AnalyticsLabel) => ({
  ...TRANSACTIONS_EVENTS.EXECUTE,
  eventLabel: label,
})
