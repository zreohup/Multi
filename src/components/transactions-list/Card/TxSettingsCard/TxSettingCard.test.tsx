import { render } from '@/src/tests/test-utils'
import { TxSettingsCard } from '.'
import { mockTransferWithInfo } from '@/src/tests/mocks'
import { TransactionInfoType } from '@safe-global/store/gateway/types'
import { SettingsChangeTransaction } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'

describe('TxSettingCard', () => {
  it('should render the default markup', () => {
    const container = render(
      <TxSettingsCard
        onPress={() => null}
        txInfo={
          mockTransferWithInfo({
            type: TransactionInfoType.SETTINGS_CHANGE,
          }) as SettingsChangeTransaction
        }
      />,
    )

    expect(container).toMatchSnapshot()
  })
})
