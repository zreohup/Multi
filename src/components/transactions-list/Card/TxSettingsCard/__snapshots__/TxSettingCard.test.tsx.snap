// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxSettingCard should render the default markup 1`] = `
<View>
  <View
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 7,
        "borderTopLeftRadius": 7,
        "borderTopRightRadius": 7,
        "flexDirection": "column",
        "flexWrap": "wrap",
        "gap": 12,
        "justifyContent": "flex-start",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 12,
            "maxWidth": "100%",
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#EEEFF0",
              "borderBottomLeftRadius": 100,
              "borderBottomRightRadius": 100,
              "borderTopLeftRadius": 100,
              "borderTopRightRadius": 100,
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#121312",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "SafeIcons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            î¥¯
          </Text>
        </View>
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 4,
                "marginBottom": 4,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#A1A3A7",
                  "fontFamily": "DM Sans",
                  "fontSize": 13,
                  "marginBottom": 2,
                }
              }
              suppressHighlighting={true}
            >
              Settings change
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#121312",
                "fontFamily": "DMSans-SemiBold",
                "fontSize": 14,
              }
            }
            suppressHighlighting={true}
          >
            mockMethod
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
